File: ProyectosCSharp/SoftProg/SoftProgBusiness/RRHH/BOI/EmpleadoBOImpl.cs
using SoftProgBusiness.RRHH.BO;
using SoftProgModel.RRHH;
using SoftProgPersistance.RRHH.DAO;
using SoftProgPersistance.RRHH.Impl;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgBusiness.RRHH.BOI
{
    public class EmpleadoBOImpl : IEmpleadoBO
    {
        private EmpleadoDAO daoEmpleado;

        public EmpleadoBOImpl()
        {
            daoEmpleado = new EmpleadoImpl();
        }
        public int eliminar(int idEmpleado)
        {
            return daoEmpleado.eliminar(idEmpleado);
        }

        public int insertar(Empleado empleado)
        {
            validar(empleado);
            return daoEmpleado.insertar(empleado);
        }

        public BindingList<Empleado> listarTodos()
        {
            return daoEmpleado.listarTodos();
        }

        public int modificar(Empleado empleado)
        {
            validar(empleado);
            return daoEmpleado.modificar(empleado);
        }

        public Empleado obtenerPorId(int idEmpleado)
        {
            return daoEmpleado.obtenerPorId(idEmpleado);
        }

        public void validar(Empleado empleado)
        {
            if (empleado.DNI.Trim().Length == 0)
                throw new Exception("El DNI del empleado no puedes estar vacío.");
            if (empleado.DNI.Trim().Length != 8)
                throw new Exception("El DNI del empleado debe ser de 8 dígitos.");
            if (empleado.Nombre.Trim().Length == 0)
                throw new Exception("El nombre del empleado no puede estar vacío.");
            if (empleado.ApellidoPaterno.Trim().Length == 0)
                throw new Exception("El apellido del empleado no puede estar vacío.");
            if (empleado.Cargo.Trim().Length == 0)
                throw new Exception("El cargo del empleado no puede estar vacío.");
            if (empleado.Sueldo <= 0)
                throw new Exception("El sueldo no puede ser un 0 ni un número negativo.");
        }
    }
}

File: Arquitectura_REST_SOAP/Backend/SoftProgBusiness/src/main/java/pe/edu/pucp/softprog/rrhh/bo/EmpleadoBOImpl.java
package pe.edu.pucp.softprog.rrhh.bo;

import java.util.ArrayList;
import pe.edu.pucp.softprog.rrhh.dao.EmpleadoDAO;
import pe.edu.pucp.softprog.rrhh.model.Empleado;
import pe.edu.pucp.softprog.rrhh.mysql.EmpleadoImpl;
import pe.edu.pucp.softprog.rrhh.boi.IEmpleadoBO;

public class EmpleadoBOImpl implements IEmpleadoBO{

    private final EmpleadoDAO daoEmpleado;
    
    public EmpleadoBOImpl(){
        daoEmpleado = new EmpleadoImpl();
    }

    @Override
    public int eliminar(int idEmpleado) throws Exception {
        return daoEmpleado.eliminar(idEmpleado);
    }
   
    @Override
    public ArrayList listarTodos() throws Exception {
        return daoEmpleado.listarTodos();
    }

    @Override
    public int insertar(Empleado empleado) throws Exception {
        return daoEmpleado.insertar(empleado);
    }

    @Override
    public int modificar(Empleado empleado) throws Exception {
        return daoEmpleado.modificar(empleado);
    }

    @Override
    public void validar(Empleado objeto) throws Exception {
        throw new UnsupportedOperationException("Not supported yet."); // Generated from nbfs://nbhost/SystemFileSystem/Templates/Classes/Code/GeneratedMethodBody
    }

    @Override
    public Empleado obtenerPorId(int idEmpleado) throws Exception {
        return daoEmpleado.obtenerPorId(idEmpleado);
    }
    
}
